src/gui/src/inspector.cpp

```diff
@@ -271,22 +333,57 @@ void Inspector::inspect(const Selected& object)
     actions_[button] = action;
   }
 
-  // connect so announcements can be made about changes
-  connect(model_,
-          SIGNAL(itemChanged(QStandardItem*)),
-          this,
-          SIGNAL(selectedItemChanged()));
+  blockSignals(false);
 }
 
 void Inspector::clicked(const QModelIndex& index)
 {
+  // QT sends both single and double clicks, so they need to be handled with a timer to
+  // be able to tell the difference
```

Comment:
There is a separate doubleClicked signal you should be able to use.

Reply:
There is `doubleClicked(const QModelIndex &index)`, and I would use it to detect the double click, but I would still need a timer to decide if a single click was just that.
https://forum.qt.io/topic/90542/click-signal-fired-when-i-double-click-is-this-normal
I tried this out and I found that you do indeed get both the single and double click events.
So when implemented with the `QTreeView`, you end up getting `clicked`, `doubleClicked`, and `clicked`.

If there is a better way, I would love to implement that since the timer is not ideal, but googling around, that seems to be the only way to tell the two apart.

Reply:
It's ugly but I don't have a better suggestion.  The only alternative  I can see is to have an explicit edit icon (eg pencil) rather than a double click but I don't think that is much better.